tests/jballerina-unit-test/src/test/resources/test-src/types/intersection/test_intersection_type.bal

```diff
@@ -155,6 +155,24 @@ function testTypeCheckingAgainstEffectiveType2() {
     assertEquality("bar", farr[3]);
 }
 
+type R1 readonly & record {|int i;|}?;
+type R2 (readonly & record {|int i;|})?;
+
+function testR1(R1 r) returns boolean {
```

Comment:
Let's add a test or two in type guard (is check in if block and then using the variable in the type narrowed context) tests also.